File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/abstractfactory/trasnport/FactoryProducer.java
package com.bigboxcode.designpattern.abstractfactory.trasnport;

public class FactoryProducer {

    public static AbstractTransportFactory getFactory(int numberOfWheels) {

        if (numberOfWheels == 2) {
            return new TwoWheelTransportFactory();
        }

        if (numberOfWheels == 4) {
            return new FourWheelTransportFactory();
        }

        return null;
    }
}

File: design-patterns/php/abstractfactory/transport/FactoryProducer.php
<?php
// FactoryProducer.php

namespace BigBoxCode\DesignPattern\AbstractFactory\Transport;

class FactoryProducer {
    public static function getFactory(int $numberOfWheels): ?AbstractTransportFactory {
        switch ($numberOfWheels) {
            case 2:
                return new TwoWheelTransportFactory();
            case 4:
                return new FourWheelTransportFactory();
        }

        return null;
    }
}